File: runtime/src/main/java/com/siimkinks/sqlitemagic/entity/EntityPersistOperationBuilder.java
package com.siimkinks.sqlitemagic.entity;

import androidx.annotation.CheckResult;
import androidx.annotation.NonNull;

interface EntityPersistOperationBuilder<R> extends EntityOperationBuilder<R> {
  /**
   * Configure this operation to ignore {@code null} values inside entities while persisting them.
   *
   * @return Operation builder
   */
  @NonNull
  @CheckResult
  R ignoreNullValues();
}

File: runtime/src/main/kotlin/com/siimkinks/sqlitemagic/entity/EntityPersistOperationBuilder.kt
package com.siimkinks.sqlitemagic.entity

import androidx.annotation.CheckResult

interface EntityPersistOperationBuilder<R> : EntityOperationBuilder<R> {
  /**
   * Configure this operation to ignore `null` values inside entities while persisting them.
   *
   * @return Operation builder
   */
  @CheckResult
  fun ignoreNullValues(): R
}
